src/engraving/dom/property.h

```diff
@@ -449,6 +449,7 @@ enum class Pid {
     SCORE_FONT,
     SYMBOLS_SIZE,
     SYMBOL_ANGLE,
+    SCORE_STYLE_PRESET,
```

Comment:
I'm starting to doubt more and more whether this should be a "property" at all.

The first question is: of what EngravingObject is it a property? Surely not of any individual element, but perhaps of a `Score` or `MasterScore`. But these classes don't work with properties at all currently, and thus their properties are also not written to a file.

Instead of with properties, Score and MasterScore work with "style" items. So, should this option be a style option? Perhaps, but also not really: that results in the weird situation that one special style item determines which style file needs to be loaded, and that causes all other style items (and if you don't watch out, even that special item itself) to be overridden.

Judging from how you are currently using this property, namely only in ScoreAccessibilitySettingsModel, it should be neither a property, nor a style item, because its value isn't stored anywhere anyway.

Reply:
Just noticed my reply was still 'pending'. Looks like I forgot to click the 'submit' button!

---

I don't think we were fully clear about what we were expecting here, partly because we weren't completely sure ourselves.

For the MVP, the style preset dropdown can be purely a UI concept; it doesn't have to exist in the score model/file at all. When the dropdown value changes, a new style would be loaded like with Format > Load Style. MuseScore would instantly "forget" that the settings came from a file rather than manually via Format > Style.


However, I think the eventual aim is to save the name of the MSN style in the score, probably as a style rather than a property, or maybe even as an attribute on the  `<Style>` element:

```XML
<Score>
    <Style preset="16mm MSN">
        <titleFontFace>Arial</titleFontFace>
        <Spatium>4</Spatium>
        <!-- more style settings -->
    </Style>
</Score>
```

If this attribute is present, that **_does not_** mean we will ignore the individual style settings in the score and load the specified MSN style file again and again whenever the score is opened.

Instead, the individual style settings will take priority for engraving purposes. The `preset` attribute would only be used to set the UI dropdown to the correct value when the score is opened.

Once we know what value to use in the dropdown, we can potentially do something more clever with it in a future version. For example, if the user clicks "Reset all styles to default", we could reset to the specified MSN default rather than to the global default. Or we can show a dialog like this:

> **Load the new 16mm MSN style?**
> This score was created in MuseScore Studio 4.5 using an old version of the 16mm MSN style. Would you like to load the new version of this style?
> [ Yes ] &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; [ No ]

However, anything beyond simply storing the preset name and using it to determine the dropdown value is outside the scope of this GSoC project.

Reply:
Question: If there is a preset but there are also some individual settings which override the stylesheet values, should the preset value still show the preset attribute value or "Custom"?

Reply:
Thanks @shoogle @cbjeukendrup for providing the insights. I just updated the PR, will further refine it as we continue this week